packages/extension/src/view/popup/stateProviders/settings/settingsProvider.tsx

```diff
@@ -0,0 +1,184 @@
+
+    if (Object.keys(sessionStorage).includes('pendingReload')) {
+      _setSettingsChanged(sessionStorage?.pendingReload);
+
+      if (Object.keys(sessionStorage).includes('allowedNumberOfTabs')) {
+        setAllowedNumberOfTabsForSettingsDisplay(
+          sessionStorage.allowedNumberOfTabs
+        );
+      } else {
+        setAllowedNumberOfTabsForSettingsDisplay(
+          currentSettings.allowedNumberOfTabs
+        );
+      }
+
+      if (Object.keys(sessionStorage).includes('isUsingCDP')) {
+        setIsUsingCDPForSettingsDisplay(sessionStorage.isUsingCDP);
+      } else {
+        setIsUsingCDPForSettingsDisplay(currentSettings.isUsingCDP);
+      }
+    } else {
+      setAllowedNumberOfTabsForSettingsDisplay(
+        currentSettings.allowedNumberOfTabs
+      );
+      setIsUsingCDPForSettingsDisplay(currentSettings.isUsingCDP);
+    }
+
+    if (Object.keys(currentSettings).includes('allowedNumberOfTabs')) {
+      setAllowedNumberOfTabs(currentSettings.allowedNumberOfTabs);
+    }
+
+    if (Object.keys(currentSettings).includes('isUsingCDP')) {
+      _setIsUsingCDP(currentSettings.isUsingCDP);
+    }
+  }, []);
+
+  const setUsingCDP = useCallback((newValue: boolean) => {
+    setIsUsingCDPForSettingsDisplay(newValue);
+    chrome.storage.session.set({
+      isUsingCDP: newValue,
+      pendingReload: true,
+    });
+  }, []);
+
+  const sessionStoreChangeListener = useCallback(
+    (changes: { [key: string]: chrome.storage.StorageChange }) => {
+      if (
+        Object.keys(changes).includes('allowedNumberOfTabs') &&
+        Object.keys(changes.allowedNumberOfTabs).includes('newValue')
+      ) {
+        setAllowedNumberOfTabs(changes?.allowedNumberOfTabs?.newValue);
+        _setSettingsChanged(true);
+      }
+
+      if (
+        Object.keys(changes).includes('isUsingCDP') &&
+        Object.keys(changes.isUsingCDP).includes('newValue')
+      ) {
+        _setIsUsingCDP(changes?.isUsingCDP?.newValue);
+        _setSettingsChanged(true);
+      }
+    },
+    []
+  );
+
+  const handleSettingsChange = useCallback(async () => {
+    if (settingsChanged) {
+      await chrome.runtime.sendMessage({
+        type: SERVICE_WORKER_TABS_RELOAD_COMMAND,
+      });
+      _setSettingsChanged(false);
+    }
+  }, [settingsChanged]);
+
+  const changeSyncStorageListener = useCallback(
+    (changes: { [key: string]: chrome.storage.StorageChange }) => {
+      if (
+        Object.keys(changes).includes('allowedNumberOfTabs') &&
+        Object.keys(changes.allowedNumberOfTabs).includes('newValue')
+      ) {
+        setAllowedNumberOfTabs(changes?.allowedNumberOfTabs?.newValue);
+      }
+
+      if (
+        Object.keys(changes).includes('isUsingCDP') &&
+        Object.keys(changes.isUsingCDP).includes('newValue')
+      ) {
+        _setIsUsingCDP(changes?.isUsingCDP?.newValue);
+      }
+    },
+    []
+  );
+
+  const setSettingsChanged = useCallback((newValue: boolean) => {
```

Comment:
Can we just expose _setSettingsChanged directly?

Reply:
Resolved in [8cf3cca](https://github.com/GoogleChromeLabs/ps-analysis-tool/pull/519/commits/8cf3cca2bf225352a90a2884d273273426270151)